Base/Python/slicer/util.py

```diff
@@ -1091,3 +1091,97 @@ def onClick(caller,event):
   up()
   interactor.SetShiftKey(0)
   interactor.SetControlKey(0)
+
+def downloadFile(url, targetFilePath):
+  """ Download ``url`` to local storage as ``targetFilePath``
+      Target file path needs to indicate the file name and extension as well
+  """
+  import os
+  import logging
+  if not os.path.exists(targetFilePath) or os.stat(targetFilePath).st_size == 0:
+    logging.info('Downloading from\n  %s\nas file\n  %s\nIt may take a few minutes...' % (url,targetFilePath))
+    try:
+      import urllib
+      urllib.urlretrieve(url, targetFilePath)
+    except Exception, e:
+      import traceback
+      traceback.print_exc()
+      logging.error('Failed to download file from ' + url)
+      return False
+  else:
+    logging.info('Requested file has been found: ' + targetFilePath)
+  return True
+
+def extractArchive(archiveFilePath, outputDir, expectedNumberOfExtractedFiles=None):
+  """ Extract file ``archiveFilePath`` into folder ``outputDir``. Number of expected files
+      unzipped may be specified in ``expectedNumberOfExtractedFiles``.
+      If folder contains the same number of files as expected (if specified), then
+        it will be assumed that unzipping has been successfully done earlier.
+  """
```

Comment:
"""Extract file ``archiveFilePath`` into folder ``outputDir``.

If ``expectedNumberOfExtractedFiles`` is specified and the number of files contained
in ``outputDir`` matches, the function returns early indicating that the archive has already
been extracted.
 """